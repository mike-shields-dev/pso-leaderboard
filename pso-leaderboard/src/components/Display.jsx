import React from "react";
import { Link } from "react-router-dom";
import style from "../styles/Display.module.css";
import PageTitleSetter from "./PageTitleSetter";

const Display = () => {
  return (
    <>
      <PageTitleSetter title="Display" />

      <div className={style.display}>
        <p>Display</p>
        <nav>
          <Link to="/">Home</Link>
          <Link to="/editor">Editor</Link>
        </nav>
      </div>
    </>
  );
};

export default Display;
