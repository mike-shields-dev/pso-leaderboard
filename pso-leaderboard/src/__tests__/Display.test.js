import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Display from "../components/Display";

const renderDisplay = () => render(<Display />, { wrapper: BrowserRouter });

describe("Display", () => {
  it("matches snapshot", () => {
    const { asFragment } = renderDisplay();

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders hyperlinks to '/home' and '/editor'", () => {
    renderDisplay();

    expect(screen.getByRole("link", { name: /home/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /editor/i })).toBeInTheDocument();
  });
});
