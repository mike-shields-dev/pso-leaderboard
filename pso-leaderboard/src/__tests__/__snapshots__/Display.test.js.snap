// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="display"
  >
    <p>
      Display
    </p>
    <nav>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/editor"
      >
        Editor
      </a>
    </nav>
  </div>
</DocumentFragment>
`;
